app/utils/StripCharUtil.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+object StripCharUtil {
+
+  def stripAll(text: String, charactersToRemove: String): String = {
```

Comment:
I find this quite confusing to read. The recursive functions inside aren't using the clearest argument names (`a`, `n`) and recursive functions in general aren't the easiest to follow. Is this over-engineered a little? As per my comment below, if you can strip out `£`, `,` and ` ` with a single line, what is this function doing? Purely removing full stops?

The guidance suggests to only allow full stops to indicate pence. So the only case there is one full stop at the end. Could you just use `stripSuffix` at the end of the validation to remove a potential full stop from the end of the value?

Reply:
Can we remove need `charactersToRemove` argument seeing as everywhere this function is called, the same value is being passed in?